kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -649,6 +650,22 @@ public Watch watch(String resourceVersion, final Watcher<T> watcher) throws Kube
       return watch;
     } catch (MalformedURLException e) {
       throw KubernetesClientException.launderThrowable(e);
+    } catch (KubernetesClientException ke) {
```

Comment:
Should we check the type of the exception? For example, there would be no point in falling back to HTTP watch in e.g. auth error. Perhaps we should only fall back if we receive a 200 instead of a 101 for example?